src/internal/common/ptr/pointer_test.go

```diff
@@ -99,3 +99,43 @@ func (suite *PointerSuite) TestVal() {
 	assert.NotEmpty(t, myInt32)
 	assert.NotEmpty(t, myInt64)
 }
+
+func (suite *PointerSuite) TestOrNow() {
+	oneMinuteAgo := time.Now().Add(-1 * time.Minute)
+	inOneMinute := time.Now().Add(1 * time.Minute)
+
+	table := []struct {
+		name        string
+		p           *time.Time
+		expectEqual bool
+	}{
+		{
+			name:        "populated value",
+			p:           &oneMinuteAgo,
+			expectEqual: true,
+		},
+		{
+			name:        "nil",
+			p:           nil,
+			expectEqual: false,
+		},
+		{
+			name:        "pointer to 0 valued time",
+			p:           &time.Time{},
+			expectEqual: false,
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+			result := ptr.OrNow(test.p)
+
+			if test.expectEqual {
+				assert.Equal(t, *test.p, result)
+			} else {
+				assert.True(t, oneMinuteAgo.Before(result), "lower bound")
```

Comment:
could you also use the `WithinDuration` assertion here?

Reply:
But then we need a transformer for that wrapper instead.  It's a transformer either way.  Just a matter of where we want the transformation to occur.  Between this change and https://github.com/alcionai/corso/pull/4481 I'll probably revisit this for cleanliness.  But I don't see the difference as a clear win at this stage.